src/librbd/api/Group.cc

```diff
@@ -1223,9 +1223,29 @@ int Group<I>::snap_list(librados::IoCtx& group_ioctx, const char *group_name,
   for (auto snap : cls_snaps) {
     snaps->push_back(
 	group_snap_info_t {
-	   snap.name,
-	   static_cast<group_snap_state_t>(snap.state)});
+          snap.name,
+          static_cast<group_snap_state_t>(snap.state)});
+  }
+  return 0;
+}
+
+template <typename I>
+int Group<I>::snap_list(librados::IoCtx& group_ioctx, const char *group_name,
```

Comment:
Would it be better to rename this to snap_list2 ?

Reply:
I think the strategy that is used for other APIs to have a single internal handler which is called for both old and new API and have the massaging needed for the old API performed in `src/librbd/librbd.cc`.  So I'd change the existing `Group::snap_list()` to return `group_snap_info2_t` and get rid of the overload that is introduced here.